pkg/apis/numaflow/v1alpha1/scale.go

```diff
@@ -31,35 +31,40 @@ type Scale struct {
 	// rate, thus less replicas. It's only effective for source vertices.
 	// +optional
-	TargetProcessingSeconds *uint32 `json:"targetProcessingSeconds,omitempty" protobuf:"varint,7,opt,name=targetProcessingSeconds"`
+	TargetProcessingSeconds *uint32 `json:"targetProcessingSeconds,omitempty" protobuf:"varint,6,opt,name=targetProcessingSeconds"`
 	// TargetBufferAvailability is used to define the target percentage of the buffer availability.
 	// A valid and meaningful value should be less than the BufferUsageLimit defined in the Edge spec (or Pipeline spec), for example, 50.
 	// It only applies to UDF and Sink vertices because only they have buffers to read.
 	// +optional
-	TargetBufferAvailability *uint32 `json:"targetBufferAvailability,omitempty" protobuf:"varint,8,opt,name=targetBufferAvailability"`
-	// ReplicasPerScale defines maximum replicas can be scaled up or down at once.
+	TargetBufferAvailability *uint32 `json:"targetBufferAvailability,omitempty" protobuf:"varint,7,opt,name=targetBufferAvailability"`
+	// DeprecatedReplicasPerScale defines maximum replicas can be scaled up or down at once.
 	// The is use to prevent too aggressive scaling operations
 	// +optional
-	ReplicasPerScale *uint32 `json:"replicasPerScale,omitempty" protobuf:"varint,9,opt,name=replicasPerScale"`
+	// Deprecated: Use ReplicasPerScaleUp and ReplicasPerScaleDown instead
+	DeprecatedReplicasPerScale *uint32 `json:"replicasPerScale,omitempty" protobuf:"varint,8,opt,name=replicasPerScale"`
 	// ScaleUpCooldownSeconds defines the cooldown seconds after a scaling operation, before a follow-up scaling up.
 	// It defaults to the CooldownSeconds if not set.
 	// +optional
-	ScaleUpCooldownSeconds *uint32 `json:"scaleUpCooldownSeconds,omitempty" protobuf:"varint,10,opt,name=scaleUpCooldownSeconds"`
+	ScaleUpCooldownSeconds *uint32 `json:"scaleUpCooldownSeconds,omitempty" protobuf:"varint,9,opt,name=scaleUpCooldownSeconds"`
 	// ScaleDownCooldownSeconds defines the cooldown seconds after a scaling operation, before a follow-up scaling down.
 	// It defaults to the CooldownSeconds if not set.
 	// +optional
-	ScaleDownCooldownSeconds *uint32 `json:"scaleDownCooldownSeconds,omitempty" protobuf:"varint,11,opt,name=scaleDownCooldownSeconds"`
+	ScaleDownCooldownSeconds *uint32 `json:"scaleDownCooldownSeconds,omitempty" protobuf:"varint,10,opt,name=scaleDownCooldownSeconds"`
+	// ReplicasPerScaleUp defines maximum replicas can be scaled up at once.
+	// The is use to prevent too aggressive scaling up operations
+	// +optional
+	ReplicasPerScaleUp *uint32 `json:"replicasPerScaleUp,omitempty" protobuf:"varint,11,opt,name=replicasPerScaleUp"`
+	// ReplicasPerScaleDown defines maximum replicas can be scaled down at once.
+	// The is use to prevent too aggressive scaling down operations
+	// +optional
+	ReplicasPerScaleDown *uint32 `json:"replicasPerScaleDown,omitempty" protobuf:"varint,12,opt,name=replicasPerScaleDown"`
```

Comment:
// ReplicasPerScaleDown defines the maximum number of replicas that can be decreased in a single scale down operation.
// This is used to prevent over aggressive scaling down operations